app/lib/discourse_automation/scripts/close_topic.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+DiscourseAutomation::Scriptable::CLOSE_TOPIC = 'close_topic'
+
+DiscourseAutomation::Scriptable.add(DiscourseAutomation::Scriptable::CLOSE_TOPIC) do
+  field :topic_to_be_closed, component: :text, required: true
+  field :message, component: :text
+  field :user, component: :user
+
+  version 1
+
+  # FIXME: copied over from pin_topic; needs some thought
+  triggerables [:point_in_time]
+
+  script do |_context, fields|
+    next unless topic_id = fields.dig('topic_to_be_closed', 'value')
+    next unless topic = Topic.find_by(id: topic_id)
+
+    message = fields.dig('message', 'value')
+    username = fields.dig('user', 'value') || Discourse.system_user.username
```

Comment:
Why don't you rely on your default value in `#close_topic` ?

Reply:
to make it more obvious that there was a default value, without having to go to the implementation in `scriptable.rb` to see the default value in the method.
Slightly less DRY but a bit more obvious. Do we want 100% DRY in cases like these?

Reply:
that's a nitpick both are fine, mostly I would try to have default at only one place, but this is fine like this, up to you.